contrib/fsl_integration/sct_plugin.py

```diff
@@ -0,0 +1,246 @@
+
+        outfilename = os.path.join(os.getcwd(), 'seg.nii.gz')
+        image = Image(outfilename)
+        overlayList.append(image)
+
+        opts = displayCtx.getOpts(image)
+        opts.cmap = 'red'
+
+
+class TabPanelSCSeg(SCTPanel):
+
+    description = """This segmentation tool is based on Deep Learning and
+    a 3D U-Net. For more information, please refer to the
+    article below.<br><br>
+    <b>Citation</b>:<br>
+    (TODO: add Charley paper)
+    """
+
+    def __init__(self, parent):
+        super(TabPanelSCSeg, self).__init__(parent=parent,
+                                            id_=wx.ID_ANY)
+        button_gm = wx.Button(self, id=wx.ID_ANY, label="Spinal Cord Segmentation")
+        button_gm.Bind(wx.EVT_BUTTON, self.onButtonSC)
+
+        lbl_contrasts = ['t1', 't2', 't2s', 'dwi']
+        self.rbox_contrast = wx.RadioBox(self, label='Select contrast:',
+                                         choices=lbl_contrasts,
+                                         majorDimension=1,
+                                         style=wx.RA_SPECIFY_ROWS)
+
+        sizer = wx.BoxSizer(wx.VERTICAL)
+        sizer.Add(self.rbox_contrast, 0, wx.ALL, 5)
+        sizer.Add(button_gm, 0, wx.ALL, 5)
+        self.sizer_h.Add(sizer)
+        self.SetSizerAndFit(self.sizer_h)
+
+    def onButtonSC(self, event):
+        selected_overlay = displayCtx.getSelectedOverlay()
+        filename_path = selected_overlay.dataSource
+        contrast = self.rbox_contrast.GetStringSelection()
+
+        base_name = os.path.basename(filename_path)
+        fname, fext = base_name.split(os.extsep, 1)
+        out_name = "{}_seg.{}".format(fname, fext)
+
+        cmd_line = "sct_deepseg_sc -i {} -c {}".format(filename_path, contrast)
+        self.call_sct_command(cmd_line)
+
+        outfilename = os.path.join(os.getcwd(), out_name)
+        image = Image(outfilename)
+        overlayList.append(image)
+
+        opts = displayCtx.getOpts(image)
+        opts.cmap = 'red'
+
+
+def run_main():
+    window = aui_manager.GetManagedWindow()
+
+    if 'SCT_DIR' not in os.environ:
+        dlg = wx.MessageDialog(window, 'Spinal Cord Toolbox (SCT) was not '
+                               'found in your system. Please check the '
+                               'installation procedure at https://github'
+                               '.com/neuropoly/spinalcordtoolbox',
+                               'SCT not found !', wx.OK | wx.ICON_INFORMATION)
+        dlg.ShowModal()
+        dlg.Destroy()
+        return
+
+    notebook = aui.AuiNotebook(window)
+    panel_gm = TabPanelGMSeg(notebook)
+    panel_sc = TabPanelSCSeg(notebook)
+
+    notebook.AddPage(panel_gm, "sct_deepseg_gm", True)
+    notebook.AddPage(panel_sc, "sct_deepseg_sc", False)
+
+    aui_manager.AddPane(notebook, 
+                        aui.AuiPaneInfo().Name("notebook_content").
+                        CenterPane().PaneBorder(False)) 
+    aui_manager.Update()
+
+
+run_main()
```

Comment:
Replace with: 
```
if __name__ == "__main__":
    run_main()
```
it would make it the file into a module.

Reply:
Note that this script is being executed by fsleyes, so the `__name__` won't be set to `__main__`. If I add this checking, the `run_main()` will not execute.

PS: `wx` isn't installed on SCT, it is part of the fsleyes.